frame/benchmarking/src/v1.rs

```diff
@@ -446,13 +446,13 @@ macro_rules! benchmarks_iter {
 					>:: [< new_call_variant_ $dispatch >] (
 						$($arg),*
 					);
-					let __benchmarked_call_encoded = $crate::frame_support::codec::Encode::encode(
+					let __benchmarked_call_encoded = $crate::frame_support::__private::codec::Encode::encode(
 						&__call
 					);
 				}: {
 					let __call_decoded = <
 						Call<T $(, $instance )?>
-						as $crate::frame_support::codec::Decode
+						as $crate::frame_support::__private::codec::Decode
```

Comment:
this feels..wrong. 

curious why we need to do it this way, and can't just pull `Encode` and `Decode` from their `codec` crate.

Reply:
To make it correct, you would need to re-export codec from `frame-benchmarking`.

Reply:
yes, benchmarking re-exports will follow in the parent issue https://github.com/paritytech/substrate/issues/14213

Reply:
Because its a macro that emits code. The emitted code cannot make the assumption that `codec` is in the `Cargo.toml` of the crate that it is being used by.  
We therefore need to ship a macro will all exports that it needs.